Support using a non-root user

kstephens- opened the pull request:
I actually needed this functionality (https://github.com/zenika-open-source/terraform-aws-cli/issues/14) for a project, so I thought I would go ahead and open a PR for it. Incidentally, this is my first PR on github, so let me know if I've done anything obviously wrong. Also, thanks for maintaining this very convenient image.

A couple of notes on changes:

The uid and gid for the user/group, as well as the user/group names, are completely arbitrary, so if you would prefer to use different ones, let me know. 

I created a home directory for the user because aws cli wants to write a credentials cache in an .aws directory in the user's home directory. I didn't find any options to change this behavior, and I'm not sure if it would be a good idea anyway.

While running the tests in the gitlab actions, I ran into what looks like a recent change by github, where they've disabled the set-env command for security reasons. This of course caused the tests to fail. To get the tests working again, I updated the variable setting to use the new format following this issue: https://github.com/actions/toolkit/issues/641.

Let me know if you have any comments or suggestions.

bgauduch commented:
Hello @kstephens- !

Thanks a lot for your interest in this project and for opening this PR !
Welcome onboard the awesome open source community by the way 👋  !

For a first PR it's perfectly explained and detailed, nothing to say 👌 
Thanks also for the Github action corrections, It was identified but I didn't had time to correct them !

I just have a suggestion : as a user was added with a home and rights on `/workspace` folder, we need to test that (I've seen in your commit that you did try 😉).

Here is the code to add in the `container-structure-tests.yaml` file that should be sufficient enough:
```yaml
fileExistenceTests:
- name: 'Check nonroot user home'
  path: '/home/nonroot'
  shouldExist: true
  permissions: 'drwxr-xr-x'
  uid: 1001
  gid: 1001
  isExecutableBy: 'group'
- name: 'Check nonroot user rights on /workspace folder'
  path: '/workspace'
  shouldExist: true
  permissions: 'drwxr-xr-x'
  uid: 1001
  gid: 1001
  isExecutableBy: 'group'
```

Another suggestion on the user & groroup id that can be set to a lower value, 1001 for instance.
The readme should also be updated to state the use of a non-root user and it's UID / GID.

bgauduch commented on Dockerfile:
```diff
@@ -49,5 +49,11 @@ COPY --from=terraform /terraform /usr/local/bin/terraform
 COPY --from=aws-cli /usr/lib/python3/dist-packages /usr/lib/python3/dist-packages
+
 WORKDIR /workspace
+RUN groupadd -g 65500 nonroot \
+  && useradd -g nonroot -m -u 65500 nonroot \
+  && chown nonroot:nonroot /workspace
```
lower values of 1001 should be sufficient enought for both UID / GID 😉

kstephens- replied:
Updated uid and gid to 1001

kstephens- commented:
Hi @bgauduch, thanks for the prompt review! I'll have the updates you suggested ready shortly. Everything should be working fine. I'm using the image in a kubernetes job that runs some aws cli commands and terraform, and that's working perfectly, but I'll also do some more testing to verify further.

kstephens- commented:
I updated the uid and gid to 1001 and added the file permission tests. All looks good. I also updated the README with info about the non-root user. I wasn't sure where to put, so I added it in the "What's Inside ?" section, which seemed somewhat reasonable. Let me know if there's a better place for it or if I'm missing something that should be included.

I also re-verified that terraform and aws cli commands work as expected. I tried it in my kubernetes job (which runs terraform and a few aws commands) and I also tried a few commands manually, and everything worked as expected.

bgauduch commented on README.md:
```diff
@@ -37,6 +37,10 @@ Tools included:
 * [Git](https://git-scm.com/) for Terraform remote module usage, see available versions on the [Debian Packages repository](https://packages.debian.org/search?suite=buster&arch=any&searchon=names&keywords=git)
 * [jq](https://stedolan.github.io/jq/) to process JSON returned by AWS, see available versions on the [Debian Packages repository](https://packages.debian.org/search?suite=buster&arch=any&searchon=names&keywords=jq)
 
+User:
+
+This image uses a non-root with a uid and gid of 1001 to conform with docker security best practices.
```
```suggestion
This image uses a non-root user with a UID and GID of 1001 to conform with docker security best practices.
```

kstephens- replied:
Updated with suggested changes

bgauduch commented:
@kstephens- Nice work, thanks a lot for your reactivity, the fix, tests and documentation update👌 

Just a last remark on the readme: as the objective is to keep it as tidy and to the point as possible, I suggested a few changes to what you added (see requested changes). You did good adding it in the "what's inside" section.

Thanks a lot for you reactivity and for the further tests you performed 👍 
All in all, very nice first PR ! Keep up the good work 🦾

bgauduch commented:
Also @kstephens- just oout of curiosity, is your K8S job used in a CI process or do you use it in another kind of process ?

kstephens- commented:
@bgauduch, no problem happy to be able to contribute! The kubernetes job is kind of a post install process for an ingress controller. It basically waits for the ingress controller service to finish setting up a load balancer and then sets up a cloudfront distribution in front of it. It actually works pretty well once I got it set up correctly.

bgauduch commented:
@kstephens- Wow nice use case ! Glad to hear this tooling image help you in the process 👌 
Looking good, merging your changes right now !

Good luck with your project, please let me know if you get any idea on how to improve / enhance this project !

bgauduch reviewed (approved):
LGTM 👌

bgauduch commented:
@kstephens- latest image with non-root user is available here: https://hub.docker.com/layers/zenika/terraform-aws-cli/latest/images/sha256-ab983135419d9c194e2adfad7f0e1b8b01548057288b3b962fd6f736b2390c65?context=explore

kstephens- commented:
@bgauduch Excellent! Glad I could help!
